bitcoin/src/pow.rs

```diff
@@ -239,6 +246,10 @@ impl Target {
     ///
     /// See [`difficulty`] for details.
     ///
+    /// # Returns
+    ///
+    /// Returns [`f64::NAN`] if `self` is zero (caused by divide by zero).
```

Comment:
In 6c642b9b309aa5da6a51face546d722dfc693929:

I believe it will actually return positive infinity.

Reply:
Yep, this says `inf`: `println!("{}", 42.0f64 / 0.0);`

Reply:
Oh, cheers. Will fix. (I read that the return value return true for `is_nan()` and made an assumption, my bad.)

Reply:
Well, infinity literally is not a number.

Reply:
FTR I was wrong
```rust
    if f64::INFINITY.is_nan() {
        println!("not a number");
    }
```
Does not print anything.